test/conformance/enqueue/urEnqueueUSMMemcpy.cpp

```diff
@@ -1,2 +1,140 @@
 // Copyright (C) 2023 Intel Corporation
 // SPDX-License-Identifier: MIT
+
+#include <uur/fixtures.h>
+#include <vector>
+
+struct urEnqueueUSMMemcpyTest : uur::urQueueTest {
+    void SetUp() override {
+        UUR_RETURN_ON_FATAL_FAILURE(urQueueTest::SetUp());
+
+        bool device_usm{false};
+        ASSERT_SUCCESS(
+            urDeviceGetInfo(device, UR_DEVICE_INFO_USM_DEVICE_SUPPORT, sizeof(bool), &device_usm, nullptr));
+
+        if (!device_usm) {
+            GTEST_SKIP_("Device USM is not supported");
+        }
+
+        ur_usm_mem_flags_t flags{};
+        ASSERT_SUCCESS(
+            urUSMDeviceAlloc(context, device, &flags, allocation_size, 0, reinterpret_cast<void **>(&device_src)));
+        ASSERT_SUCCESS(
+            urUSMDeviceAlloc(context, device, &flags, allocation_size, 0, reinterpret_cast<void **>(&device_dst)));
+
+        ASSERT_SUCCESS(
+            urEnqueueUSMMemset(queue, device_src, memset_value, allocation_size, 0, nullptr, &memset_event));
+        ASSERT_SUCCESS(urQueueFlush(queue));
+    }
+
+    void TearDown() override {
+        if (memset_event) {
+            EXPECT_SUCCESS(urEventRelease(memset_event));
+        }
+        if (device_src) {
+            EXPECT_SUCCESS(urUSMFree(context, device_src));
+        }
+        if (device_dst) {
+            EXPECT_SUCCESS(urUSMFree(context, device_dst));
+        }
+
+        UUR_RETURN_ON_FATAL_FAILURE(urQueueTest::TearDown());
+    }
+
+    bool verifyData() {
+        EXPECT_SUCCESS(
+            urEnqueueUSMMemcpy(queue, true, host_mem.data(), device_dst, allocation_size, 0, nullptr, nullptr));
+        return std::all_of(host_mem.begin(), host_mem.end(), [this](uint8_t i) { return i == memset_value; });
+    }
+
+    const uint32_t num_elements = 1024;
+    const uint8_t memset_value = 12;
+    const uint32_t allocation_size = sizeof(uint8_t) * num_elements;
+    std::vector<uint8_t> host_mem = std::vector<uint8_t>(num_elements);
+
+    ur_event_handle_t memset_event = nullptr;
+    int *device_src{nullptr};
+    int *device_dst{nullptr};
+};
+
+/**
+ * Test that urEnqueueUSMMemcpy blocks when the blocking parameter is set to true.
+ */
+TEST_P(urEnqueueUSMMemcpyTest, Blocking) {
+
+    ASSERT_SUCCESS(urEventWait(1, &memset_event));
+    ASSERT_SUCCESS(urEnqueueUSMMemcpy(queue, true, device_dst, device_src, allocation_size, 0, nullptr, nullptr));
+    ASSERT_TRUE(verifyData());
```

Comment:
I think it make sense to assert that the event has actually completed before verifying the data:

```cpp
ur_event_status_t memset_event_status;
ASSERT_SUCCESS(urEventGetInfo(memset_event, UR_EVENT_INFO_COMMAND_EXECUTION_STATUS,
    sizeof(ur_event_status_t), &memset_event_status, nullptr));
ASSERT_EQ(UR_EVENT_STATUS_COMPLETE, memset_event_status);
```

This also applies to the other tests which verify the data.

Reply:
Done :)